.github/workflows/blocked-prs.yml

```
@@ -0,0 +1,245 @@
+name: Blocked/Stacked Pull Requests Automation
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - reopened
+      - edited
+      - synchronize
+  workflow_dispatch:
+    inputs:
+      pr_id:
+        description: Local Pull Request number to work on
+        required: true
+        type: number
+
+jobs:
+  blocked_status:
+    name: Check Blocked Status
+    runs-on: ubuntu-latest
+
+    permissions:
+      issues: write
+      pull-requests: write
+      statuses: write
+      checks: write
+
+    steps:
+      - name: Checkout Default Branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+      - name: Setup From Pull Request Event
+        if: github.event_name != 'workflow_dispatch'
+        id: pr_event_setup
+        env:
+          PR_JSON: "${{ toJSON(github.event.pull_request) }}"
+        run: |
+          # setup env for the rest of the workflow
```

```suggestion
      - name: Setup
        id: pr_event_setup
        run: |
          # setup env for the rest of the workflow
          PR_JSON="${PR_JSON:-{{ toJSON(github.event.pull_request) }}}"
```
And then do this to have one step regardless of event

---

I don't follow, won't github complain about invalid templates if `toJSON(github.event.pull_request)` fails to evaluate?